docs/intro/overrides.rst

```diff
@@ -0,0 +1,349 @@
+replace the` **same** `Page Object`.
+
+However, let's suppose that there are some :class:`~.OverrideRule` which actually
+result in a conflict. To give an example, let's suppose that rules **#2** and **#3**
+`intends to replace the` **same** `Page Object`. It would look something like:
+
+.. code-block:: python
+
+    # 2. OverrideRule(for_patterns=Patterns(include=['site_2.com'], exclude=[], priority=500), use=<class 'ecommerce_page_objects.site_2.EcomSite2'>, instead_of=<class 'common_items.ProductGenericPage'>, meta={})
+    # 3. OverrideRule(for_patterns=Patterns(include=['site_2.com'], exclude=[], priority=500), use=<class 'gadget_sites_page_objects.site_2.GadgetSite2'>, instead_of=<class 'common_items.ProductGenericPage'>, meta={})
+
+Notice that the ``instead_of`` param are the same and only the ``use`` param
+remained different.
+
+There are two main ways we recommend in solving this.
+
+**1. Priority Resolution**
+
+If you notice, the ``for_patterns`` attribute of :class:`~.OverrideRule` is an
+instance of `url_matcher.Patterns
+<https://url-matcher.readthedocs.io/en/stable/api_reference.html#module-url-matcher>`_.
+This instance also has a ``priority`` param where a higher value will be chosen
+in times of conflict.
+
+.. note::
+
+    The `url-matcher`_ library is the one responsible breaking such ``priority`` conflicts
+    `(amongst others)`. It's specifically discussed in this section: `rules-conflict-resolution
+    <https://url-matcher.readthedocs.io/en/stable/intro.html#rules-conflict-resolution>`_.
+
+Unfortunately, updating the ``priority`` value directly isn't possible as the
+:class:`url_matcher.Patterns` is a **frozen** `dataclass`. The same is true for
+:class:`~.OverrideRule`. This is made by design so that they are hashable and could
+be deduplicated immediately without consequences of them changing in value.
+
+The only way that the ``priority`` value can be changed is by creating a new
+:class:`~.OverrideRule` with a different ``priority`` value (`higher if it needs
+more priority`). You don't necessarily need to `delete` the **old**
+:class:`~.OverrideRule` since they will be resolved via ``priority`` anyways.
+
+If the conflict cannot be resolved by the ``priority`` param, then
+the next approach could be used.
+
+**2. Specifically Selecting the Rules**
+
+When the last resort of ``priority``-resolution doesn't work, then you could always
+specifically select the list of :class:`~.OverrideRule` you want to use.
+
+We **recommend** in creating an **inclusion**-list rather than an **exclusion**-list
+since the latter is quite brittle. For instance, an external package you're using
+has updated its rules and the exlusion strategy misses out on a few rules that
+were recently added. This could lead to a `silent-error` of receiving a different
+set of rules than expected.
+
+For this approach, you can use the :meth:`~.PageObjectRegistry.search_overrides`
```

Comment:
Hm, I was thinking about importing POs directly. How would you do inclusion in search_overrides?

Reply:
Added another in-depth example for `search_overrides()` in https://github.com/scrapinghub/web-poet/pull/27/commits/74573319c7b495dbf16dc615be846af51721d508.